src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -758,6 +759,16 @@ private static DataContract CreateGetOnlyCollectionDataContract(int id, RuntimeT
                 return dataContract;
             }
 
+            private static Type GetDataContractAdapterTypeForGeneratedAssembly(Type type)
+            {
+                if (type == Globals.TypeOfDateTimeOffset)
```

Comment:
Nit: could you explain in a comment why this method is needed and why DateTimeOffset is the only type getting special treatment?  From the name it sounds like a very general purpose method, but it is only mapping this single type.